### PR TITLE
Improve check-pypi script

### LINKED ISSUE #446: Consider adding a flag to `komodo-check-pypi` to check only minor/patch version changes
Currently the script will suggest the upgrade to the newest available tag, while it could be nice to be able to choose whether you need major upgrades or not

### LINKED ISSUE #447: check-pypi script showing os incompatible upgrades
check-pypi for packages might suggest packages that are incompatible with the target OS / platform. An example of this is PyQt5-Qt5. The current latest version is 5.15.12. We are currently using 5.15.2, so naturally it suggests upgrading this. Upon building the new release on a linux machine, it failed because it could not find the package. Turns out, 5.15.12 only has wheels for macosx.

check-pypi script should take the platform or OS in as an argument, or default on the OS/platform of the host running the script, and filter out the upgrades not compatible.

### PR DESCRIPTION
**Pull request by jonathan-eq**
Resolves #447
Resolves #446

### CODE REVIEW

File: komodo/check_up_to_date_pypi.py [hunk 1]
```diff
@@ -281,6 +292,15 @@ def get_args() -> argparse.Namespace:
             "If given, will ignore packages with names that match the regex expression"
         ),
     )
+    parser.add_argument(
+        "--target-platform",
+        default=sys.platform,
+        choices=["darwin", "win32", "linux"],
+        help=(
+            "Which OS the new package version should be compatible with. Should be either linux,"
+            " macos, or win32. Defaults to OS of host machine"
```
Comment (kvashchuka) [inline komodo/check_up_to_date_pypi.py:329]: small thing - the choices do not match the help

File: komodo/check_up_to_date_pypi.py [hunk 2]
```diff
@@ -56,7 +57,19 @@ def get_python_requirement(sources: list):
     return ""
 
 
-def compatible_versions(releases: dict, python_version):
+def is_platform_compatible(build_info: List[Dict], platform: str) -> bool:
+    if platform.lower() == "darwin":
+        platform = "macos"
```
Comment (kvashchuka) [inline komodo/check_up_to_date_pypi.py:62]: Why do we need this `if`? Could you explain, I do not follow

Comment (jonathan-eq) [reply komodo/check_up_to_date_pypi.py:62]: sys.platform on mac returns darwin, but it should be macos in the filename, so I convert it to macos

File: komodo/check_up_to_date_pypi.py (hunk 1 above)
Comment (jonathan-eq) [reply komodo/check_up_to_date_pypi.py]: Fixed!

**Review by kvashchuka — APPROVED** [review]
good job!

**Pull request merged**